.github/workflows/BuildPushJava.yaml

```diff
@@ -0,0 +1,181 @@
+name: Code Build & Image Push (CICDv2)
+permissions: write-all
+
+on:
+  workflow_call:
+    inputs:
+      project-id:
+        default: 'gocariq-platform'
+        type: string
+      app-name:
+        required: true
+        type: string
+      slack-channel:
+        default: '#release'
+        type: string
+      deploy-env:
+        required: true
+        type: string
+      registry-name:
+        default: 'cariq'
+        type: string
+      image-tag:
+        default: '0.1.0'
+        type: string
+      sonar:
+        default: 'https://sonar.gocariq.com'
+        type: string
+    secrets:
+      SLACK_WEBHOOK_URL:
+        required: true
+      SONAR_TOKEN:
+        required: true
+      GOOGLE_CRED:
+        required: true
+      REGISTRY_HOST_NAME:
+        required: true
+      GKE_SA_GHA_PUSHER_CREDENTIALS:
+        required: true
+      GHA_DEVOPS_DISPATCH:
+        required: true
+      K8S_POD_DEPLOY_VALIDATION:
+        required: true
+      GHA_DISPATCH_TEST:
+        required: true
+jobs:
+  build:
+    runs-on: ubuntu-22.04
+    steps:
+
+      - name: Print input variables
+        run: |
+          echo "Project ID: ${{ inputs.project-id }}"
+          echo "App name: ${{ inputs.app-name }}"
+          echo "Slack channel: ${{ inputs.slack-channel }}"
+          echo "Deploy environment: ${{ inputs.deploy-env }}"
+          echo "Registry name: ${{ inputs.registry-name }}"
+          echo "Image tag: ${{ inputs.image-tag }}"
+
+      - uses: act10ns/slack@v2.0.0
+        with:
+          status: starting
+          channel: ${{ inputs.slack-channel }}
+        if: always()
+
+      - uses: actions/checkout@v3
+
+      - name: Image check
+        id: check-image
+        uses: gocariq-gitops/release-actions/checkDockerImage@reusable_workflows
+        with:
+          google_credentials: ${{secrets.GOOGLE_CRED}}
+          google_project_id: ${{ inputs.project-id }}
+          docker_repo_name:  ${{ inputs.registry-name }}
+          docker_image_name: ${{ inputs.app-name }}
+
+      - name: test echo
+        run: |
+          echo ${{ env.IMAGE_EXIST }}
+          echo ${{ env.SHORT_SHA }}
+
+      - name: Dockerfile lint
+        if: ${{ env.IMAGE_EXIST == 'false' }}
+        id: validate-dockerfile
+        shell: bash
+        run: |
+          docker run --rm -e HADOLINT_IGNORE=DL3007,DL3008,DL3018 -i hadolint/hadolint < Dockerfile
+
+      - name: helm lint
+        if: ${{ env.IMAGE_EXIST == 'false' }}
+        id: validate-chart
+        run: |
+          helm lint ./charts/${{ inputs.app-name }}/
+
+      - name: Set up JDK 11
+        if: ${{ env.IMAGE_EXIST == 'false' }}
+        id: setting-env
+        uses: actions/setup-java@v2
+        with:
+          java-version: 11
```

Comment:
we can user version as env variable I think we can also reference it in the step name so we do not need to use hard coded value in the step name

Reply:
This step is going to be removed in future